src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -157,4 +157,23 @@ public function urlRedirectAction(Request $request, string $path, bool $permanen
 
         return new RedirectResponse($url, $statusCode);
     }
+
+    public function __invoke(Request $request): Response
+    {
+        $p = $request->attributes->get('_route_params', []);
+
+        if (\array_key_exists('route', $p)) {
+            if (\array_key_exists('path', $p)) {
+                throw new \RuntimeException('Ambiguous redirection settings, use the "path" or "route" parameter, not both.');
+            }
+
+            return $this->redirectAction($request, $p['route'], $p['permanent'] ?? false, $p['ignoreAttributes'] ?? false, $p['keepRequestMethod'] ?? false, $p['keepQueryParams'] ?? false);
```

Comment:
for all the common parameters, you could make them arguments of the controller instead, to let Symfony bind them (with any logic applied by argument resolver, as done for other methods) and take the default values from the method.

Reply:
I'm not sure if it's worth it just for two common parameters.